docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/07_ReactJS_Redux_and_GraphQL.md

```diff
@@ -795,4 +795,511 @@ You could manipulate the action called by the originalReducer, there isn't an ex
+
+Lastly, let's just register all this with `Injector`.
+
+*myapp/client/registerDependencies.js*
+```js
+//...
+import AddForm from './AddForm';
+import createNote from './createNote';
+
+const registerDependencies = () => {
+  //...
+  Injector.component.register('NoteAddForm', AddForm);
+  Injector.query.register('CreateNote', createNote);
+
+  //...
+  Injector.transform(
+    'notesaddform-graphql',
+    (updater) => {
+      updater.component('NoteAddForm', injectGraphql('CreateNote'));
+    }
+  );
+};
+
+export default registerDependencies;
+```
+
+This is exactly the same pattern as we did before with a query, only with different components and GraphQL abstractions this time. Note that even though `CreateNote` is a mutation, it still gets registered under `Injector.query` for simplicity.
+
+### Extending mutations
+
+Now let's switch back to our thirdparty module that is customising our Notes application. The developer is going to want to ensure that users can supply a "Priority" value for each note entered. This will involve updating the `AddForm` component.
+
+*my-other-app/client/transformAddForm.js*
+```js
+import React from 'react';
+
+const transformAddForm = () => ({ onAdd }) => {
+  let content, priority;
+  return (
+    <div>
+      <label>Note content</label>
+      <input type="text" ref={node => content = node}/>
+      <label>Priority</label>
+      <select ref={node => priority = node}>
+        <option value="low">Low</option>
+        <option value="medium">Medium</option>
+        <option value="high">High</option>
+      </select>
+      <button onClick={(e) => {
+        e.preventDefault();
+        if (content && priority) {
+          onAdd(content.value, priority.value);
+        }
+      }}>Add</button>
+    </div>
+  );
+};
+
+export default transformAddForm;
+```
+
+We've extended the `onAdd` callback to take two parameters -- one for the note content, and another for the priority. We'll need to update the mutation to reflect this.
+
+*my-other-app/client/transformCreateNote.js*
+```js
+const transformCreateNote = (manager) => {
+  manager.addField('Priority');
+  manager.transformApolloConfig('props', ({ mutate }) => (prevProps) => {
+    const onAdd = (content, priority) => {
+      mutate({
+        variables: {
+          Input: {
+            Content: content,
+            Priority: priority
+          }
+        }
+      });
+    };
+
+    return {
+      ...prevProps,
+      onAdd,
+    };
+  })
+};
+
+export default transformCreateNote;
+```
+
+All we've done here is overridden the `props` setting in the `CreateNote` apollo config. Recall from the previous section that it maps the `mutate` function to the `onAdd` prop. Since we've changed the signature of that function, we'll need to override the entire prop.
+
+Now we just need to register these transforms, and we're done!
```

Comment:
would be a good idea to note the importance of spreading the `prevProps` at the end :)

easily missed and you lose previous transformations/properties if forgotten.

Reply:
@unclecheese can you action this please?